deployer/deploy_functions.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"go/types"
+	"log"
+	"os"
+	"strings"
+	"text/template"
+
+	"golang.org/x/tools/go/packages"
+)
+
+func main() {
+	argsWithoutProg := os.Args[1:]
+	if len(argsWithoutProg) != 1 {
+		fmt.Println("Usage: go run deploy_functions.go <FunctionPackage>")
+		os.Exit(1)
+	}
+
+	pkg := argsWithoutProg[0]
+	triggers, err := extractTriggers(pkg)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	code, err := generateEntrypoint(pkg, triggers)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	fmt.Println(code)
+}
+
+type Trigger struct {
+	Name string
+	Type string
+}
+
+const httpTriggerType = "github.com/inlined/go-functions/https.Function"
+const pubsubTriggerType = "github.com/inlined/go-functions/pubsub.Function"
+
+var triggerTypes = []string{
+	httpTriggerType,
+	pubsubTriggerType,
+}
+
+func extractTriggers(pkg string) ([]Trigger, error) {
+	exports, err := loadExports(pkg)
+	if err != nil {
+		return nil, err
+	}
+
+	var triggers []Trigger
+	for _, exp := range exports {
+		trig := Trigger{
+			Name: exp.Name(),
+			Type: exp.Type().String(),
+		}
+		triggers = append(triggers, trig)
+	}
+
+	return triggers, nil
+}
+
+func loadExports(pkg string) ([]types.Object, error) {
+	cfg := &packages.Config{Mode: packages.LoadTypes}
+	pkgs, err := packages.Load(cfg, pkg)
+	if err != nil {
+		return nil, err
+	}
+	if packages.PrintErrors(pkgs) > 0 {
+		return nil, fmt.Errorf("one or mor errors")
+	}
+
+	var results []types.Object
+	scope := pkgs[0].Types.Scope()
+	for _, symbol := range scope.Names() {
+		typeInfo := scope.Lookup(symbol)
+		if isTrigger(symbol, typeInfo) {
```

Comment:
While we clearly need to make sure we don't add an unexported type, we don't really need to do the typecheck against pubsub/http functions. emulator.Serve will already ignore unsupported types and using a filter like this means that we would have to remember to update the codegen code every time we add a new trigger.

Reply:
Done. This makes code even more simple.